src/hotspot/share/code/nmethod.cpp

```diff
@@ -112,7 +112,7 @@
 // Cast from int value to narrow type
 #define CHECKED_CAST(result, T, thing)      \
   result = static_cast<T>(thing); \
-  assert(static_cast<int>(result) == thing, "failed: %d != %d", static_cast<int>(result), thing);
+  guarantee(static_cast<int>(result) == thing, "failed: %d != %d", static_cast<int>(result), thing);
```

Comment:
This check is not is a hot path and so afford to be a guarantee. That said, I've no strong objection to leaving it as an assert.